test/testother.cpp

```diff
@@ -4807,6 +4815,26 @@ class TestOther : public TestFixture {
             "}");
         ASSERT_EQUALS("", errout.str());
 
+        check(
+            "bool foo(unsigned x) {\n"
+            "  int y = 0;\n"
+            "  if (x < y)"
+            "    return true;\n"
+            "  return false;\n"
+            "}");
+        ASSERT_EQUALS("[test.cpp:3]: (style) Checking if unsigned expression 'x' is less than zero.\n", errout.str());
```

Comment:
I guess there are situations where it will not be obvious that RHS is 0 when looking at the code and then the warning should explain why Cppcheck thinks so. I suggest that you use `Check::getErrorPath` somehow when reporting this warning.

I would say that when the value is a integer literal then it is preferable to not use `Check::getErrorPath`.

Reply:
We should have some utility function somewhere that can determine if an expression is a "literal". However I do not find it. The `isConstExpr` sounds good but does not do it. The `isConstVarExpr` is pretty close, it's just a shame that it returns true for constant variables. I will try to find something...

Reply:
> I would say that when the value is a integer literal then it is preferable to not use Check::getErrorPath.

I am not sure but maybe you can ignore this advice. Does it generate good output if you always use `Check::getErrorPath`?

Reply:
```
#define z 0
void f(unsigned x) {
    if (x < 0) {}
}
void g(unsigned x) {
    int y = 0;
    if (x < y) {}
}

void h(unsigned x) {
    if (x < z) {}
}
```
With `Check::getErrorPath`:
```
cppcheck --enable=style unsigned-example.c
Checking unsigned-example.c ...
[unsigned-example.c:3]: (style) Checking if unsigned expression 'x' is less than zero.
[unsigned-example.c:7]: (style) Checking if unsigned expression 'x' is less than zero.
[unsigned-example.c:11]: (style) Checking if unsigned expression 'x' is less than zero.

cppcheck --enable=style --verbose unsigned-example.c
Checking unsigned-example.c ...
Defines:
Undefines:
Includes:
Platform:Native
[unsigned-example.c:3]: (style) The unsigned expression 'x' will never be negative so it is either pointless or an error to check if it is.
[unsigned-example.c:6] -> [unsigned-example.c:7]: (style) The unsigned expression 'x' will never be negative so it is either pointless or an error to check if it is.
[unsigned-example.c:11]: (style) The unsigned expression 'x' will never be negative so it is either pointless or an error to check if it is.
```
So yes, the output if there is a "literal" seems to fine. However, in order to get the full "path" in other cases, one has to specify `--verbose` (see below). Would that be sufficient?

The condition for getting the full path is (verbose I understood what it does, I didn't check the other two):
```
    (mSettings->verbose || mSettings->xml || !mSettings->templateLocation.empty())
```

Reply:
`mSettings->xml` is xml output I suppose. Then I think this should be ok behaviour (it's the same way the zerodiv-checker works).

Btw, do you prefer me to squash the changes before I push or do you prefer 'delta-commits' (for simpler reviewing?).